v2/yarpcgrpc/headers.go

```diff
@@ -66,17 +66,13 @@ const (
 	EncodingHeader = "rpc-encoding"
 	// ErrorNameHeader is the header key for the error name.
 	ErrorNameHeader = "rpc-error-name"
+	// ErrorDetailsHeader is the header key for the error details used in gRPC.
+	// https://sourcegraph.com/github.com/grpc/grpc-go/-/blob/internal/transport/http_util.go#L317-326
+	ErrorDetailsHeader = "grpc-status-details-bin"
 	// ApplicationErrorHeader is the header key that will contain a non-empty value
 	// if there was an application error.
 	ApplicationErrorHeader = "rpc-application-error"
```

Comment:
We can remove ApplicationErrorHeader and ApplicationErrorHeaderValue safely, or at least leave their behavior unchanged and orthogonal to the grpc status details binary header, for compatibility with existing YARPC gRPC services. We need to engage with the gRPC community separately to find a different header than grpc-status-details-bin for non-Protobuf error details. We don’t have to validate gRPC+Thrift yet so we can leave this unimplemented for now.